File: java/src/test/java/it/gabrieletondi/telldontaskkata/doubles/InMemoryProductCatalog.java
package it.gabrieletondi.telldontaskkata.doubles;

import it.gabrieletondi.telldontaskkata.domain.Product;
import it.gabrieletondi.telldontaskkata.repository.ProductCatalog;

import java.util.List;

public class InMemoryProductCatalog implements ProductCatalog {
    private final List<Product> products;

    public InMemoryProductCatalog(List<Product> products) {
        this.products = products;
    }

    public Product getByName(final String name) {
        return products.stream().filter(p -> p.getName().equals(name)).findFirst().orElse(null);
    }
}

File: kotlin/src/test/java/it/gabrieletondi/telldontaskkata/doubles/InMemoryProductCatalog.kt
package it.gabrieletondi.telldontaskkata.doubles

import it.gabrieletondi.telldontaskkata.domain.Product
import it.gabrieletondi.telldontaskkata.repository.ProductCatalog

class InMemoryProductCatalog(private val products: List<Product>) : ProductCatalog {
    override fun getByName(name: String): Product? {
        return products.stream().filter { p: Product -> p.name == name }.findFirst().orElse(null)
    }
}